### PR TITLE
doc: simplify browsing and understanding of java extension

### LINKED ISSUE #81: documentation is really hard to browse
**Problem**: I never find what I want on revapi.org

**High-level cause**: the link structure is counter-intuitive, some links are somehow hidden

**Low-level causes**: 

* the top-level menu is dynamic (sometimes it changes with a new "Project Details")
* the project details are visible only upon click (hence one does not find elements with Ctrl-F)
* the "Table of Contents" refers to anchors of the same page and not to other pages of the same sub-project

**Tentative solutions**:

Solution A:
* the top-level menu is stable (always the same items)
* the "Project Details" are put as a new flat horizontal menu below the top menu

Solution B:
* the top-level menu is stable (always the same items)
* the "Project Details" are put on the right hand side, at the same place as the (Anchor) Table of contents
* The  (Anchor) Table of contents is moved at the top or bottom of the page

### PR DESCRIPTION
**Pull request by monperrus**
fix #81 

this PR does for Java what was done in #82 for the maven plugin

### CODE REVIEW

File: revapi-java/src/site/asciidoc/index.adoc [hunk 1]
```diff
@@ -58,7 +58,2479 @@ class formally part of the API even though it comes from a dependency. The `Mode
...
+
+A method might no longer be visible to code that used to call it. This is a breaking change.
+
+[options=header]
+|====
+2+| Match parameters
+| `package` | the package of the class
+| `classSimpleName` | the simple name of the class
+| `methodName` | the name of the method added
+| `oldVisibility` | The visibility of the type as it was in the old version.
```
Comment (metlos) [inline revapi-java/src/site/asciidoc/index.adoc]: This link should not be mentioned (and shouldn't have been before your changes either 😉  ).

File: revapi-java/src/site/asciidoc/enhance-java-checks.adoc [hunk 2]
```diff
@@ -1,116 +0,0 @@
-== Enhancing Java API checks
```
Comment (metlos) [inline revapi-java/src/site/asciidoc/enhance-java-checks.adoc]: Where did you move this file?

Comment (monperrus) [reply revapi-java/src/site/asciidoc/enhance-java-checks.adoc]: is now in index.adoc

**Comment by monperrus** [comment]
> This link should not be mentioned (and shouldn't have been before your changes either 😉 ).
>
what link? I don't see one here?

**Comment by monperrus** [comment]
pushed changes according to your comments

File: revapi-java/src/site/asciidoc/index.adoc [hunk 3]
```diff
@@ -58,7 +58,2596 @@ class formally part of the API even though it comes from a dependency. The `Mode
...
```
Comment (metlos) [inline revapi-java/src/site/asciidoc/index.adoc]: This link should not be here (and shouldn't have been here for some time)

Comment (metlos) [reply revapi-java/src/site/asciidoc/index.adoc]: WTH?

Comment (metlos) [reply revapi-java/src/site/asciidoc/index.adoc]: This comment should be at line 2218, but github seems to disagree with me everytime I add it on that line...

**Comment by monperrus** [comment]
done
--Marti

**Pull request merged**